src/job-exporter/src/nvidia.py

```diff
@@ -69,26 +69,29 @@ class NvidiaGpuStatus(object):
         gpu_util the gpu util of this gpu, float number, range 0~100
         gpu_mem_util the gpu memory usage/total of this gpu, float number, range 0~100
         pids an array of pid numbers that uses this card
-        ecc_errors instance of EccError class """
-    def __init__(self, gpu_util, gpu_mem_util, pids, ecc_errors, minor, uuid):
+        ecc_errors instance of EccError class
+        temperature will be None or float celsius """
+    def __init__(self, gpu_util, gpu_mem_util, pids, ecc_errors, minor, uuid, temperature):
         self.gpu_util = gpu_util # float
         self.gpu_mem_util = gpu_mem_util # float
         self.pids = pids # list of int
         self.ecc_errors = ecc_errors # list of EccError
         self.minor = minor
         self.uuid = uuid # str
+        self.temperature = temperature # None or float celsius
 
     def __repr__(self):
-        return "util: %.3f, mem_util: %.3f, pids: %s, ecc: %s, minor: %s, uuid: %s" % \
-                (self.gpu_util, self.gpu_mem_util, self.pids, self.ecc_errors, self.minor, self.uuid)
+        return "util: %.3f, mem_util: %.3f, pids: %s, ecc: %s, minor: %s, uuid: %s, temperature %.3f" % \
+                (self.gpu_util, self.gpu_mem_util, self.pids, self.ecc_errors, self.minor, self.uuid, self.temperature)
 
-    def __eq__(self, o):
+    def __eq__(self, o): # for test
         return self.gpu_util == o.gpu_util and \
                 self.gpu_mem_util == o.gpu_mem_util and \
                 self.pids == o.pids and \
                 self.ecc_errors == o.ecc_errors and \
                 self.minor == o.minor and \
-                self.uuid == o.uuid
+                self.uuid == o.uuid and \
+                self.temperature == o.temperature
 
 
 def parse_smi_xml_result(smi):
```

Comment:
It's better to use [NVIDIA System Management Interface (NVML)](https://developer.nvidia.com/nvidia-system-management-interface), or its [Python bindings]( http://pypi.python.org/pypi/nvidia-ml-py/) to get GPU status, which is backwards compatible, while `nvidia-smi`'s output is not guaranteed to be backwards compatible according to its docs.

Reply:
Ok, will try to investigate and use them in future PR.